### DISCUSSION TITLE
Cleanup old, large logfiles

### DISCUSSION THREAD

**Discussion by ZonPV** [johanmeijer/grott#185] [Q&A]
I run grott version 2.7.6 in a docker environment on my Synology and collect lots of daily logfiles. It gives a time-out when I try to have a look in them.
Had in mind that in earlier versions there was no problem.

![Logs Grott](https://user-images.githubusercontent.com/11488827/194285402-37873677-3a06-4e20-bb0d-2da0981d28b0.JPG)

Is there a way to clean them up, or is there a setting? Just a few days back in the past will do for me.

**Comment by johanmeijer** [comment]
I am afraid this is a Synology thing. Nothing changed to grott logging the last releases

**Reply by ZonPV** [reply]
I don't think that a host interact with services in a container which runs in Docker..

In which folder are the logs written? 
Is there a setting in Grott.ini (possible) to redirect the logfiles to the host?
And how are they deleted? I guess by crontab?

**Comment by johanmeijer** [comment]
Grott writes its data standard to $stdout. It is to docker (or the docker implementation on the Synology) what to with this. The Grott container / image does not have a lot to say about that I think. 

My RPI docker implementation keeps logging to the same docker file.  Which I can see with:  docker logs grott. This can keeps the log of days (months). Stop / start the container does not clean the logs. Only rebuilding the container (remove / run) cleans the log.

I still have to found out how to control this more. It seems that Synology has created something for this, looking at your menu structure.

**Reply by ZonPV** [reply]
Up till now, I found out that there is no job or service which cleans logfiles. There is no crontab at all for any user. 
Would be nice to use parameters for loglevels for Grott; like debug, critical, warning or info...
A usual log of a usual day counts around 38000 rows a day...  

Is it a good idea to create an issue?

I can try to create a crontab for that in the existing container but have to do that after every upgrade. That makes not much sense..
You wrote: _My RPI docker implementation keeps logging to the same docker file. Which I can see with: docker logs grott._
I can use that command on Synology too. The syntax on the CLI is exact the same. Maybe

**Comment by johanmeijer** [comment]
Grott has a verbose function that is standard enable in the Container. The way implement it at this moment that it can not be disabled (simply because I use the command line -v verbose switch during start up and that overrides al other parameter/environmental setting.

I will change that in the next version of the docker Image. 

In the mean while maybe we can found oud how we can limited the log size of a docker container. I see there are possibilities for this in the parameters of docker itself and also during start up of a container. Maybe there are some experienced docker users who read this and can help us with that.
